src/tracing/service/tracing_service_impl.cc

```diff
@@ -4140,6 +4142,40 @@ base::Status TracingServiceImpl::FlushAndCloneSession(
     return PERFETTO_SVC_ERR("Not allowed to clone a session from another UID");
   }
 
+  if (session->write_into_file) {
+    if (!args.output_file_fd) {
+      return PERFETTO_SVC_ERR(
+          "Failed to clone 'write_into_file' session: a file descriptor is "
+          "required to copy existing file");
+    }
+    // TODO(ktimofeev): write comment about two different clone options
+    // (read/don't read from file), Traceur and how it is related to the
+    // 'buffer_clone_preserve_read_iter' flag (see b/382209797).
+    if (!base::flags::buffer_clone_preserve_read_iter) {
+#if !PERFETTO_BUILDFLAG(PERFETTO_OS_WIN)
+      // On Windows 'base::GetFileSize' expects 'PlatformHandle', but
+      // 'ScopedFile::operator*()' returns int, so the next line won't compile.
+      // This code never executes on Windows, so we just '#ifdef' it.
+      bool session_write_into_file_empty_file =
+          base::GetFileSize(*session->write_into_file).value_or(-1) <= 0;
+      if (!session_write_into_file_empty_file) {
+        return PERFETTO_SVC_ERR(
+            "Failed to clone 'write_into_file' session: flag "
+            "'buffer_clone_preserve_read_iter=true' required");
+      }
+#endif
+    }
+    base::FlushFile(*session->write_into_file);
+    base::Status status =
+        base::CopyFileContents(*session->write_into_file, *args.output_file_fd);
+    if (!status.ok()) {
+      return PERFETTO_SVC_ERR(
+          "Failed to clone 'write_into_file' session: failed to copy existing "
+          "file: %s",
+          status.c_message());
+    }
+  }
```

Comment:
isn't this missing an else, like

if (original_session.write_into_file) {

  ...
} else {
  args.output_file_fd.reset()?  //// !this
}